cmake/ProjectSettings.cmake

```diff
@@ -0,0 +1,71 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright 2013-2022 The Foundry Visionmongers Ltd
+
+#-----------------------------------------------------------------------
+# Set a default build type if none was specified
+
+if (NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CONFIGURATION_TYPES)
+    message(STATUS "Setting build type to 'Release' as none was specified.")
+    set(CMAKE_BUILD_TYPE Release CACHE STRING "Choose the type of build." FORCE)
+    # Set the possible values of build type for cmake-gui, ccmake
+    set_property(
+        CACHE CMAKE_BUILD_TYPE
+        PROPERTY STRINGS
+        "Debug"
+        "Release"
+        "MinSizeRel"
+        "RelWithDebInfo")
+endif ()
+
+
+#-----------------------------------------------------------------------
+# Default install directory
+
+# If the user hasn't configured cmake with an explicit
+# CMAKE_INSTALL_PREFIX, then set it to safely install into ./dist, to
+# help prevent the user from accidentally writing over /usr/local or
+# whatever.
+if (CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT AND PROJECT_IS_TOP_LEVEL)
+    set(CMAKE_INSTALL_PREFIX
+        "${PROJECT_SOURCE_DIR}/dist" CACHE PATH "Installation location" FORCE)
+endif ()
+
+
+#-----------------------------------------------------------------------
+# Interprocedural Optimization
+
+option(ENABLE_IPO "Enable Interprocedural Optimization, aka Link Time Optimization (LTO)" OFF)
```

Comment:
It sets a global variable that will apply to everything, so seemed best placed here. This file was mostly lifted from [a previous version](https://github.com/cpp-best-practices/cpp_starter_project/blob/dfd6956418ed3a298770ccc362184574879d1b33/cmake/StandardProjectSettings.cmake) of https://github.com/cpp-best-practices/cpp_starter_project  (been massively restructured in just the last few days, grr).

It can be set per-target though, so I'll move it out of here.

Reply:
Enabled IPO per-target rather than globally, as well as enabled by default (googling suggests there's no good reason to disable it in a project of this size), in 9b15fbbc